providers/infoblox/infoblox.go

```diff
@@ -272,3 +276,60 @@ func (d *InfobloxProvider) setRecordType(ty string, records []*Record) []*Record
 	}
 	return records
 }
+
+func (d *InfobloxProvider) SendRequest(method, urlStr, body string, head map[string]string) ([]*Record, error) {
+	if urlStr == "" {
+		return nil, fmt.Errorf("SendRequest to infoblox url is nil")
+	}
+
+	if method == "GET" {
+		// Using pagination on all get requests
+		// Checking if needs '?' or '&'
+		var url string
+		if strings.Contains(urlStr, "?") {
+			url = urlStr + "&" + paginate
+		} else {
+			url = urlStr + "?" + paginate
+		}
+		logrus.Debugf("SendRequest to infoblox with pagination: [method]%s, [url] %s", method, url)
+
+		res, err := d.client.SendRequest(method, url, body, head)
+		if err != nil {
+			return nil, err
+		}
+
+		var result ResultPagination
+		if err = res.Parse(&result); err != nil {
+			return nil, err
+		}
+		logrus.Debugf("SendRequest infoblox get %d of %s records", len(result.Result), maxResults)
+		records := result.Result
+		for result.Page_id != "" {
+			// Adding _page_id to nextPage url
+			nextPage := url + "&_page_id=" + result.Page_id
+			logrus.Debugf("SendRequest infoblox getting next _page_id %s", result.Page_id)
+			// emptying result.Page_id
+			result.Page_id = ""
+
+			res, err := d.client.SendRequest(method, nextPage, body, head)
+			if err != nil {
+				continue
```

Comment:
why do we continue on err? Wouldn't error mean that something went wrong, and we should return it?

Reply:
Thinking partial result, but you are right. Returning error.